src/jvm/main/org/jetbrains/kotlinx/lincheck/LinChecker.kt

```diff
@@ -87,6 +94,12 @@ class LinChecker (private val testClass: Class<*>, options: Options<*, *>?) {
         return null
     }
 
+    private fun lacksFreeMemory(): Boolean {
+        val runtime = Runtime.getRuntime()
```

Comment:
Wait, the warning is no longer here...

Reply:
This is not a warning, just a safety mechanism which ensures that `OutOfMemoryError` can not be caused by verifier. Typically, 100 iterations will pass faster than the memory will be exhausted. Yes, GC can be postponed until necessary but Garbage Collectors still try to avoid situations of "almost completely filled memory" since they can lead to long pauses. False positive alarm can lead to reduced performance (for which we do not have experiments), but false negative alarm leads to `OutOfMemoryError` and to Lincheck crash